app/bundles/UserBundle/Tests/Functional/UserPreferenceTest.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Mautic\UserBundle\Tests\Functional;
+
+use Mautic\CoreBundle\Test\MauticMysqlTestCase;
+use Mautic\UserBundle\Entity\Role;
+use Mautic\UserBundle\Entity\User;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\PasswordHasher\PasswordHasherInterface;
+
+/**
+ * Functional tests for user preference functionality.
+ *
+ * This test class verifies that:
+ * - Users can successfully set their preferences via API endpoint
+ * - Preferences are properly stored in the database
+ * - The system responds with appropriate success messages
+ *
+ * Specifically tests the column visibility preference for contacts as a representative example.
+ */
+class UserPreferenceTest extends MauticMysqlTestCase
+{
+    public function testUserCanSetPreference(): void
+    {
+        $user = $this->createTestUser();
+
+        $this->loginUser($user);
+        $this->client->setServerParameter('PHP_AUTH_USER', $user->getUsername());
+        $this->client->setServerParameter('PHP_AUTH_PW', 'Maut1cR0cks!');
+
+        $this->client->request(
+            Request::METHOD_POST,
+            '/s/user/set-preference',
+            [],
+            [],
+            ['CONTENT_TYPE' => 'application/json'],
+            json_encode([
+                'preference' => 'user_column_visibility_contacts',
+                'value'      => ['name', 'email', 'id'],
+            ])
+        );
+
+        $response = $this->client->getResponse();
+
+        $this->assertEquals(Response::HTTP_OK, $response->getStatusCode(), $response->getContent());
```

Comment:
```suggestion
        $this->assertResponseIsSuccessful();
```
this does the same thing and adds some more debug info if fails.